flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/StringWriter.java

```diff
@@ -82,7 +87,7 @@ public void open(FileSystem fs, Path path) throws IOException {
 	public void write(T element) throws IOException {
 		FSDataOutputStream outputStream = getStream();
 		outputStream.write(element.toString().getBytes(charset));
-		outputStream.write('\n');
+		outputStream.write(rowDelimiter.getBytes(charset));
```

Comment:
We could store the `delimiter` in bytes as the private field in `StringWriter.java` instead of serializing this every time.

Reply:
ok,done

Reply:
getCharsetName()  method is protected,is it better to be public  @Myasuka

Reply:
test has been passed

Reply:
Any other suggestions? @Myasuka

Reply:
I move this  to a new pr，https://github.com/apache/flink/pull/8621 ！Thanks very much！

Reply:
Sorry for late reply, I noticed that you have addressed all my suggestions on the new PR.